ingress/controllers/nginx/configuration.md

```diff
@@ -0,0 +1,394 @@
+## Contents
+* [Customizing NGINX](#customizing-nginx)
+* [Custom NGINX configuration](#custom-nginx-configuration)
+* [Custom NGINX template](#custom-nginx-template)
+* [Annotations](#annotations)
+* [Custom NGINX upstream checks](#custom-nginx-upstream-checks)
+* [Authentication](#authentication)
+* [Rewrite](#rewrite)
+* [Rate limiting](#rate-limiting)
+* [Secure NGINX upstreams](#secure-nginx-upstreams)
```

Comment:
I can't think of a better term for upstreams, backends maybe? something more generic. I wouldn't know what upstreams is if I only use haproyx.